### PR TITLE
Feature/kotlin practicals

### PR DESCRIPTION
**Pull request by BrijeshBarasiya**
Basic_of_Kotlin

### CODE REVIEW

**Comment by BrijeshBarasiya** [comment]
Done_with_BasicOfKotlin

File: app/src/main/java/com/example/basicofkotlin/AbstractAndEnumClass.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.example.basicofkotlin
+
+//Enumerations, Abstract Class and Constractor
+fun main()
+{
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/AbstractAndEnumClass.kt:5]: Opening brace should be at the same line of function declaration

File: app/src/main/java/com/example/basicofkotlin/AbstractAndEnumClass.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+        else
+            println("Wrong Input!!")
+    }
+}
+enum class Subject //Creating Enumerations Class witth Name Subject
+{
+    xCode,Java,Python,C,Dotnet,Kotlin,Php,Hyml,CSS //Defining constants values
+}
+abstract class StudentDetails //Created abstract class with name StudentDetails
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/AbstractAndEnumClass.kt:28]: Keep spacing between entities

File: app/src/main/java/com/example/basicofkotlin/ClassObject.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }catch (e: ArithmeticException){ //Handler
+        println(e)
+        check = true
+    }catch (e: NumberFormatException){ //Handler
+        println(e)
+        check = true
+    }catch (e: IllegalAccessException){//Handler
+        println(e)
+        check = true
+    }catch (e: NullPointerException){ //Handler
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ClassObject.kt:23]: } catch (e: NullPointerException) {

File: app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    println("\nIndex of B is: ${arrayList.indexOf("B")}") //Printing Index of Element B
+    arrayList.clear() //Clearing arrayList (ArrayList will empty)
+    print("Size of List is: " + arrayList.size) //Printing Size of ArrayList
+
+
+    //List
+    println("\nWelcome to List")
+    var simpleList = listOf("A","B","C","D","E") //Created New List
+    var i:Int =0
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt:47]: var i: Int = 0

File: app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    city.filter { it.startsWith("A") }.sortedBy { it }.map { it.uppercase() }.forEach { println(it) }
+
+
+    //DoWhile
+    println("\nWelcome to DoWhileLoop")
+    var count:Int = 0
+    do {
+        print(count)
+        count++
+    }while (count == 5)
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt:63]: spacing

File: app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        println("item at $i is ${simpleList[i]}")
+        i++
+    }
+    val city = listOf("Rajkot", "Ahmadabad", "Surat","Anand", "Gandhinagar","Junagadh","Ankleshwar")
+    city.filter { it.startsWith("A") }.sortedBy { it }.map { it.uppercase() }.forEach { println(it) }
+
+
+    //DoWhile
+    println("\nWelcome to DoWhileLoop")
+    var count:Int = 0
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt:59]: spacing

File: app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        print("$i ")
+    print("\n")
+
+    for(i in 50 downTo 0 step 3) //Reverse Loop and Dcrement by 3
+        print("$i ")
+
+
+    //MutableList
+    println("\nWelcome to MutableList")
+    val number = mutableListOf(1,2,3,4,5,6,7,8,9)
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ListAndLoop.kt:82]: spacing 
val number = mutableListOf( 1, 2, 3, 4......

File: app/src/main/java/com/example/basicofkotlin/BasicCode.kt [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package com.example.basicofkotlin
+
+import java.util.*
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/BasicCode.kt:3]: Do not use wildcard imports, always use specified imports

File: app/src/main/java/com/example/basicofkotlin/ClassObject.kt [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    finally {
+        if(check)
+            println("Error Genrated")
+        else
+            println("No any Error Genrated")
+    }
+}
+
+open class AdditionSubtraction{ //Creating New Class with name AdditionSubtraction
+    fun addition(value1: Int, value2: Int): Int{ //Created Function in AdditionSubtraction Class
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/ClassObject.kt:39]: new line between class declaration and first function declaration

**Pull request merged**